### ISSUE TITLE
[1.7.10] Large nether portals causes crash (U_D7-U_D8)

### ISSUE THREAD

**Issue by Ghostlyr** [sp614x/optifine#703]
Fenced the home territory with giant portals and here is the result:

```
java.lang.ArrayIndexOutOfBoundsException: -177
	at ChunkCacheOF.func_147439_a(ChunkCacheOF.java:90)
	at net.minecraft.block.BlockPortal.func_149646_a(SourceFile:136)
	at net.minecraft.client.renderer.RenderBlocks.func_147736_d(RenderBlocks.java:6786)
	at net.minecraft.client.renderer.RenderBlocks.func_147784_q(RenderBlocks.java:4788)
	at net.minecraft.client.renderer.RenderBlocks.func_147805_b(RenderBlocks.java:327)
	at net.minecraft.client.renderer.WorldRenderer.func_147892_a(WorldRenderer.java:325)
	at net.minecraft.client.renderer.RenderGlobal.func_72716_a(RenderGlobal.java:2379)
	at net.minecraft.client.renderer.EntityRenderer.func_78471_a(EntityRenderer.java:1657)
	at net.minecraft.client.renderer.EntityRenderer.func_78480_b(EntityRenderer.java:1392)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1001)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
```
full: https://pastebin.com/a96G4FLa
same issue here: http://www.minecraftforum.net/forums/support/modded-client-support/2819597-my-modded-minecraft-world-crashes
PS: Default graphic settings

**Comment by sp614x** [comment]
Can't reproduce it so far.
Does it also happen in vanilla (no mods).
World download link?

**Comment by Ghostlyr** [comment]
Sorry for the late response.
Unfortunately I can not give a map and add something about the problem, excluding [mod-list](https://pastebin.com/9wJRzTLx) and the fact that it occurred on the [Thermos b57](https://github.com/CyberdyneCC/Thermos/releases) server. And also that on version 1.7.10_HD_U_C1 problem was not observed.

**Comment by sp614x** [comment]
This should be fixed in OptiFine 1.7.10_HD_U_E3 and newer.
